### PR TITLE
Merge 2017 back into master

### PR DESCRIPTION
**Pull request by etanshaul**
A couple known issues still need to be resolved (work in progress):

1) IJ test framework's more agressive leak detection in 2017 is getting triggered again by something introduced in master. I think its isolated to the new tests in the flex/standard deployment editor test classes.

2) The flex / standard deployment UI is slightly misaligned: the service and version fields are pushed far to the right

I've done some initial sanity check testing, but this will require more extensive testing to validate.

### CODE REVIEW

**Comment by etanshaul** [comment]
What would be very helpful for validation:

@patflynn - validate that the Kokoro configs are correct for master

@nbashirbello & @patflynn - give the plugin a test drive, checking critical paths introduced in both 2017(/17.2) and master

File: kokoro/common.cfg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (patflynn) [inline kokoro/common.cfg]: these kokoro config files stay out. They've been moved.

File: gradle.properties [hunk 2]
```diff
@@ -15,8 +15,8 @@
 #
 
 ideaEdition = IC
```
Comment (patflynn) [inline gradle.properties]: what do you think about default to IU?

**Comment by etanshaul** [comment]
also, I just realized that this loses all the commits from 2017 and we want to maintain that history. Let me see what I can do ...

File: kokoro/ubuntu/release.sh [hunk 3]
```diff
@@ -59,27 +59,21 @@ if [ "$SOURCE_VERSION" != "$TAG_VERSION" ]
...
 
-echo "Installing aktau/github-release.."
+echo "Installing github-release"
 go get github.com/itchio/gothub
 cd github/google-cloud-intellij
 echo "Building plugins"
 ./gradlew buildPlugin
 
 echo "Creating Github release for tag: $GIT_TAG_NAME"
-
```
Comment (patflynn) [inline kokoro/ubuntu/release.sh]: we want to keep release.sh as is. This is actually a newer version than the branch.

**Comment by etanshaul** [comment]
I may have shot myself in the foot here by not thinking about the need to preserve the commit history from 2017 until after doing this merge. Best case I can back out and force push to this branch; worst case I have to do this merge over again. I'll experiment with a few different approaches.

**Comment by etanshaul** [comment]
Ok, I finally fixed it I think. The history from 2017 is back. Ill continue validating this merge and making fixes from code review.

**Comment by etanshaul** [comment]
Reminder to self:

When merging this in after approval, make sure to disable squashing and enable merge commits so that master gets all the commit history hear.

Also, I can squash individual commits related to review into 1 "merge 2017 to master" commit.

File: kokoro/common.cfg (hunk 1 above)
Comment (etanshaul) [inline kokoro/common.cfg]: which configs specifically need to be removed? just common.cfg?

File: kokoro/ubuntu/release.sh (hunk 3 above)
Comment (etanshaul) [inline kokoro/ubuntu/release.sh]: I don't think I follow this comment - are you saying the version in 2017 is newer than this? What needs to change exactly?

File: gradle.properties (hunk 2 above)
Comment (etanshaul) [inline gradle.properties]: I'll do this in a different PR

File: kokoro/ubuntu/release.sh (hunk 3 above)
Comment (etanshaul) [inline kokoro/ubuntu/release.sh]: I think maybe the version in master is incorrect:
the echo statement says its pull aktau/github-releases, but then the command pulls from itchio/gothub. 

Also, should I have kepts all those exports and --name lines below?

File: kokoro/common.cfg (hunk 1 above)
Comment (patflynn) [inline kokoro/common.cfg]: all the kokoro cfg and script file changes should be removed. Master has the latest config.

**Comment by etanshaul** [comment]
@patflynn - done, I've reverted all the Kokoro changes in the merge. Its now the same as master.

**Pull request merged**